RawVideo/RawVideo/ContentView.swift

```diff
@@ -0,0 +1,1032 @@
+    @State private var incomingVideoStreamRendererView: RendererView?
+    @State private var outgoingPixelBuffer: CVPixelBuffer?
+    @State private var incomingPixelBuffer: CVPixelBuffer?
+    @State private var w: Double = 0.0
+    @State private var h: Double = 0.0
+    @State private var framerate: Float = 30.0
+    @State private var maxWidth: Double = 1920.0;
+    @State private var maxHeight: Double = 1080.0;
+    @State private var token: String = "ACS token"
+    @State private var meetingLink: String = "Teams meeting link"
+    @State private var callInProgress: Bool = false
+    @State private var showCallSettings: Bool = false
+    @State private var loading: Bool = false
+    
+    struct CameraItem : Hashable
+    {
+        var id: Int
+        var name: String
+    }
+    
+    struct VideoDeviceInfoItem : Hashable
+    {
+        var id: Int
+        var name: String
+    }
+    
+    enum OutgoingVideoStreamTypeItem : String, CaseIterable
+    {
+        case localOutgoing
+        case virtualOutgoing
+        case screenShareOutgoing
+        
+        func ToVideoStreamType() -> VideoStreamType
+        {
+            switch self
+            {
+            case .localOutgoing:
+                return VideoStreamType.localOutgoing
+            case .virtualOutgoing:
+                return VideoStreamType.virtualOutgoing
+            case .screenShareOutgoing:
+                return VideoStreamType.screenShareOutgoing
+            }
+        }
+    }
+    
+    enum IncomingVideoStreamTypeItem : String, CaseIterable
+    {
+        case remoteIncoming
+        case rawIncoming
+        
+        func ToVideoStreamType() -> VideoStreamType
+        {
+            switch self
+            {
+                case .remoteIncoming:
+                    return VideoStreamType.remoteIncoming
+                case .rawIncoming:
+                    return VideoStreamType.rawIncoming
+            }
+        }
+    }
+    
+    var body: some View {
+        GeometryReader { geometryReader in
+            ZStack(alignment: .center) {
+                if (loading)
+                {
+                    ProgressView()
+                        .zIndex(1)
+                }
+                
+                VStack {
+                    if (!showCallSettings)
+                    {
+                        VStack {
+                            TextEditor(text: $token)
+                            .frame(width: 250, height: 30)
+                            .padding(10)
+                            .overlay(RoundedRectangle(cornerRadius: 5)
+                                .stroke(Color.black, lineWidth: 2))
+                            .focused($isFocused)
+                            .onChange(of: token) { _ in
+                                if token.last?.isNewline == .some(true)
+                                {
+                                    token.removeLast()
+                                    isFocused = false
```

Comment:
If we want to trim, we can use `token.trimmingCharacters(in: .whitespacesAndNewlines)`